File: src/main/java/io/github/hiiragi283/material/HTMaterialsCore.java
package io.github.hiiragi283.material;

import java.util.Collections;
import java.util.List;
import java.util.Map;

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;

import org.jetbrains.annotations.Nullable;

import zone.rong.mixinbooter.IEarlyMixinLoader;

@IFMLLoadingPlugin.Name(HMReference.MOD_NAME)
@IFMLLoadingPlugin.MCVersion("1.12.2")
public final class HTMaterialsCore implements IFMLLoadingPlugin, IEarlyMixinLoader {

    // IFMLLoadingPlugin //

    @Override
    public String[] getASMTransformerClass() {
        return new String[] {};
    }

    @Override
    public String getModContainerClass() {
        return null;
    }

    @Nullable
    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) {}

    @Override
    public String getAccessTransformerClass() {
        return null;
    }

    // IEarlyMixinLoader //

    @Override
    public List<String> getMixinConfigs() {
        return Collections.singletonList("mixins.ht_materials.json");
    }
}

File: src/main/kotlin/io/github/hiiragi283/common/core/HTMaterialsCore.kt
package io.github.hiiragi283.common.core

import io.github.hiiragi283.api.HTMaterialsAPI
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin
import zone.rong.mixinbooter.IEarlyMixinLoader

@IFMLLoadingPlugin.Name(HTMaterialsAPI.MOD_NAME)
@IFMLLoadingPlugin.MCVersion("1.12.2")
class HTMaterialsCore : IFMLLoadingPlugin, IEarlyMixinLoader {

    //    IFMLLoadingPlugin    //

    override fun getASMTransformerClass(): Array<String> = arrayOf()

    override fun getModContainerClass(): String? = null

    override fun getSetupClass(): String? = null

    override fun injectData(data: MutableMap<String, Any>) = Unit

    override fun getAccessTransformerClass(): String? = null

    //    IEarlyMixinLoader    //

    override fun getMixinConfigs(): List<String> = listOf("mixins.ht_materials.json")
}